app/src/common/shared/com/igalia/wolvic/utils/DictionaryUtils.java

```diff
@@ -0,0 +1,206 @@
+package com.igalia.wolvic.utils;
+
+import android.content.Context;
+
+import androidx.annotation.NonNull;
+import androidx.annotation.Nullable;
+
+import com.igalia.wolvic.BuildConfig;
+import com.igalia.wolvic.browser.SettingsStore;
+
+import java.io.File;
+import java.util.Arrays;
+import java.util.Comparator;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+public class DictionaryUtils {
+    public static final String EXTERNAL_DICS_SUFFIX = "_wordlist.db";
+    public static final String BUILTIN_DICS_PREFIX = "databases/";
+    public static final Map<String, String[]> BUILTIN_DICS_MAP = Map.of("zh_CN", new String[]{"google_pinyin.db"}, "zh_TW", new String[]{"zhuyin_words.db", "zhuyin_phrases.db"});
+
+    /**
+     * Returns a path in the database path that stores the remote dictionaries.
+     *
+     * @param context An activity context.
+     * @param lang    The dictionary language. This maps to the Remote properties JSON "lang" dictionary property
+     *                for external dictionaries and the dictionary value list in option_value.xml for builtin ones.
+     * @return The location of the dictionary in the devices memory.
+     */
+    @Nullable
+    public static String getExternalDicPath(@NonNull Context context, @NonNull String lang) {
+        File file = new File(context.getDatabasePath(getExternalDicFullName(lang)).getAbsolutePath());
+        if (file.exists() && file.isFile()) return file.getAbsolutePath();
+        return null;
```

Comment:
Mind reversing the logic here? As the method is getExternalDicPath I think it's better if we return null if `!file.exists() || !file.isFile()` and then after that an unconditional `return file.getAbsolutePath()`. It's mainly an aesthetic change but helps to quickly identify what's the expected/default behaviour